### ISSUE TITLE
Rename the Exec plugin to eliminate confusion with the 'waypoint exec' command

### ISSUE THREAD

**Issue by jennbriden** [hashicorp/waypoint#1079]
**Is your feature request related to a problem? Please describe.**
The [Exec plugin](https://www.waypointproject.io/plugins/exec) is used as a type of escape hatch to be able to use Waypoint with existing app workflows and still have the benefit of the Waypoint Entrypoint functions. When you use the plugin, you reference it within your Waypoint app config file (waypoint.hcl). 

Conversely, `[waypoint exec](https://www.waypointproject.io/docs/exec)` (run at the command line), is used to open a shell into a running application that you have deployed with Waypoint.

These are clearly very different functions, and I find it confusing that we use the same name both places. It does not make sense to rename the command/function, so it has to be the plugin. 

**Describe the solution you'd like**
Rename the Exec plugin to something other than "Exec." The name should convey that the plugin is an escape hatch that calls out to an external tool or system. 

**Describe alternatives you've considered**
Potential new names for the Exec plugin (please recommend others):

external
remote
out
??

**Explain any additional use-cases**
NA

**Additional context**
https://discuss.hashicorp.com/t/name-confusion-between-the-waypoint-exec-command-and-the-waypoint-exec-plugin/20598

**Comment by jgrau** [comment]
I agree that there's a name clash and it's a source of confusion. A quick thought on an alternative name - although I'm not sure it completely removes the confusion - would be `local-exec`. The argument being that people familiar with terraform will understand that it's a command that is being run on the local machine.

**Comment by jennbriden** [comment]
I like the idea of **local-exec** since it runs as part of the` deploy` step.

**Comment by briancain** [comment]
I think something like `local-exec` is close, but it will have its own kind of confusion. From the TF docs:

> The local-exec provisioner invokes a local executable after a resource is created. This invokes a process on the **machine running Terraform, not on the resource**

In this case, running `waypoint exec` isn't going to exec something locally, but on the instance that was deployed

But perhaps there's something like `something-exec` in that nature that would separate the naming confusion between the CLI command and the Exec plugin.

**Comment by jgrau** [comment]
> In this case, running `waypoint exec` isn't going to exec something locally, but on the instance that was deployed

If I understand this issue correctly then the proposal is to rename the *plugin* not the *command*. So the `something-exec` will actually be run on the local machine and not on the remote instance.

**Comment by briancain** [comment]
@jgrau - Ah, yes. See, the confusion here already exists :sweat_smile: For renaming the plugin, which will run commands locally, `Local-Exec` makes sense to me.

**Comment by jennbriden** [comment]
> > In this case, running `waypoint exec` isn't going to exec something locally, but on the instance that was deployed
> 
> If I understand this issue correctly then the proposal is to rename the _plugin_ not the _command_. So the `something-exec` will actually be run on the local machine and not on the remote instance.

Yes, that's what I was proposing. :D

**Comment by izaaklauer** [comment]
We don't believe that the effort of renaming this is worth breaking backward compat by changing the name for existing plugins. We will update the docs to clarify this.
